### DISCUSSION TITLE
Updates about migration to C#

### DISCUSSION THREAD

**Discussion by mjishnu** [mjishnu/alt-app-installer#68] [General]
the first paragraph is mostly gonna be kinda of a dev blog so you can skip if you dont wanna hear me rambling

it been a few months since i decided to do this at the time i was really new to C# and winui, i was learning and building then i got busy and distracted after i made the home/search result ui, so after few months of break i am back now that i got a few day off due university study leave.

currently most of the UI parts (home page,search result page, bundle page) are done i wll update what's completed as time goes.

![Screenshot 2024-11-18 133114](https://github.com/user-attachments/assets/9e3a3d29-6f9a-4df4-aeec-333fe2453c5c)

![Screenshot 2024-11-18 133127](https://github.com/user-attachments/assets/ad53bad9-ec67-404f-997e-9e4556a9f40a)

![Screenshot 2024-11-18 133122](https://github.com/user-attachments/assets/b6f3304f-0797-471b-9ede-efdabfac3f3e)

if you want a video demo of current progress here is the gdrive [link](https://drive.google.com/file/d/1eP2fOoddtQ6z4lq8g2Y_yirBEr8ZdIrY/view?usp=sharing)
(light mode in the demo video is distorted due to AMD Vari bright bug in the latest win 11, the page looks like the screenshots in light mode not like the one in the video)

**Comment by mjishnu** [comment]
there are currently 2 design for search bar 

left search bar:
![left search (1)](https://github.com/user-attachments/assets/b9e450b3-f79a-4ec1-b3f5-d4124a1013b7)
![left search (2)](https://github.com/user-attachments/assets/5eb05eee-9fce-4395-bf23-4ca7ed779228)
<img width="366" alt="left search(3)" src="https://github.com/user-attachments/assets/00e3f4c3-e88a-466f-b6df-409bc2b45af2" />
<img width="366" alt="left search" src="https://github.com/user-attachments/assets/e5afb325-ab3e-4711-9ada-ce5c21702a13" />

top search bar:
![top search (1)](https://github.com/user-attachments/assets/62fa2713-c68f-40bf-9e7b-5248985abc2c)
![top search (4)](https://github.com/user-attachments/assets/e36ce385-2615-4eb0-8955-5ae7f84fe753)
![top search (2)](https://github.com/user-attachments/assets/dc9f06f4-5888-4490-934a-80a3cc780f6a)
![top search (3)](https://github.com/user-attachments/assets/ea9e28b4-1cdc-4bd7-be9f-875ff6c36355)

which one should we implement tell your suggestion?

**Reply by moraroy** [reply]
Left

**Comment by azhe403** [comment]
Top

**Comment by devcombiscoito** [comment]
Definitely

**Comment by mjishnu** [comment]
<img width="1902" height="988" alt="Screenshot 2026-02-10 222359" src="https://github.com/user-attachments/assets/dca7e923-eaba-4ebe-8a17-2dc43ef47970" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222408" src="https://github.com/user-attachments/assets/35f36470-d46f-488a-ae76-e6f90cc00f51" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222424" src="https://github.com/user-attachments/assets/bbf19a94-5d15-40c1-8120-61224ccf9fa5" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222452" src="https://github.com/user-attachments/assets/76c05fdb-108c-45fb-867e-9856996ffd0d" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222515" src="https://github.com/user-attachments/assets/dd448bec-a34c-478e-82b0-67fc1cff7584" />
<img width="2560" height="1368" alt="Screenshot 2026-02-10 222527" src="https://github.com/user-attachments/assets/441ee070-b803-4f20-bfa8-600a683ec1b4" />
<img width="2560" height="1368" alt="Screenshot 2026-02-10 222538" src="https://github.com/user-attachments/assets/aa61d425-7768-417e-b61a-322e3c0ad0af" />
<img width="2560" height="1368" alt="Screenshot 2026-02-10 222602" src="https://github.com/user-attachments/assets/f7c95f03-1e49-49fd-a056-519e2a1317ba" />
<img width="2560" height="1368" alt="Screenshot 2026-02-10 222612" src="https://github.com/user-attachments/assets/3e175f66-0b8c-4927-9e99-3281eaeb4093" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222249" src="https://github.com/user-attachments/assets/b3d2c105-fb45-4e70-8c47-573d83a5db33" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222302" src="https://github.com/user-attachments/assets/79d51cbd-1952-46ed-a850-b9b16a981325" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222312" src="https://github.com/user-attachments/assets/6bb1b11a-0625-499a-8d2c-606c41446ede" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222327" src="https://github.com/user-attachments/assets/6c2fd4cc-137a-4a89-b866-66ede28aa113" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222334" src="https://github.com/user-attachments/assets/34c0a65f-b158-4aec-ac45-1eda79ab0ae3" />
<img width="2560" height="1368" alt="image" src="https://github.com/user-attachments/assets/fa981407-28a5-4dd5-9b3a-6266bc7bc780" />
<img width="1902" height="988" alt="Screenshot 2026-02-10 222350" src="https://github.com/user-attachments/assets/d591c4fc-3ba5-42e1-a653-41bd31519e5f" />

here are some current screenshots the app is 70-80% completed a few more changes maybe release in a few months no eta might take longer.

- currently working on updates detection? if anyone have any good ideas on how to do this anyhelp is welcomed

completed:
- main app functionality search, app page, downloads page, install page
- detect installed apps show open instead of install
- delta update file (download only the changed part for updates rather than the whole file)
- auto close background running apps and install no more cant close the app errors while installing 
- auto complete in search bar
- and many more like cached downloads etc.

remaining todos:
- download only option, reinstall option
- ignore dependency filters option
- create update page
- update detection for all installed apps efficiently
- support translations

Also the name has been chosen as Raven as per the poll!

**Comment by mjishnu** [comment]
implementation completed check it out in https://github.com/mjishnu/Raven 
thanks for the feedbacks!
